mlir/tools/mlir-miopen-lib/mlir-miopen-lib.cpp

```diff
@@ -122,53 +143,105 @@ extern "C" MlirHandle CreateMlirHandle(const char *arguments) {
         strToInt("dilation_h"), strToInt("dilation_w"),
         strToInt("conv_stride_h"), strToInt("conv_stride_w"),
         strToInt("padding_h"), strToInt("padding_w"), module, builder,
-        argMap["kernel_name"], mlir::FloatType::getF32(&(handle->context)),
-        false);
-
-    if (argMap["arch"] != "")
-      s_arch = argMap["arch"];
+        argMap["kernel_name"], type, false);
   }
 
   return handle;
 }
 
-extern "C" void MlirLowerCpp(MlirHandle mlirHandle) {
-  MlirHandle_s *handle = static_cast<MlirHandle_s *>(mlirHandle);
+extern "C" MlirmiopenStatus
+mlirmiopenDestroyHandle(MlirmiopenHandle mlirHandle) {
+  MlirmiopenHandle_s *handle = static_cast<MlirmiopenHandle_s *>(mlirHandle);
+  if (handle == nullptr)
+    return MlirmiopenInvalidParam;
+
+  delete handle;
+  return MlirmiopenSuccess;
+}
+
+extern "C" MlirmiopenStatus
+mlirmiopenGetExecutionDims(MlirmiopenHandle mlirHandle, size_t *global_size,
```

Comment:
@sjw36 It just occurred to me that we have not test routine that exercise this path. Is it possible to add some minimal testing to make sure this API can work? For example, I think you can add this in `mlir-miopen-lib-test.cpp` just to make sure return code is 0 and global/local sizes are non-zero.